gpt4all-chat/chatgpt.cpp

```diff
@@ -121,7 +121,7 @@ void ChatGPT::prompt(const std::string &prompt,
     QJsonDocument doc(root);
 
 #if defined(DEBUG)
-    qDebug() << "ChatGPT::prompt begin network request" << qPrintable(doc.toJson());
```

Comment:
Can you explain?

Reply:
[qPrintable](https://doc.qt.io/qt-5/qtglobal.html#qPrintable) is used when you need a `const char *` for use with e.g. printf. This is not needed for qDebug(), which has native support for the QString type. See the added line below your comment.

Reply:
The reason it was used here is because qDebug prints a QString with extra formatting and for readability I wanted to use qPrintable.

Reply:
You're right. That's a strange API... these should be changed to qUtf8Printable for [correctness](https://doc.qt.io/qt-5/qtglobal.html#qPrintable). Can I do that here or should it happen in a separate PR?

Reply:
i mean we can use noquote, but then the other strings/classes I don't mind the way that qDebug formats them in this case and no quote would ruin those

Reply:
[QDebug::noquote](https://doc.qt.io/qt-5/qdebug.html#noquote) only applies to QChar, QString, and QByteArray. These three examples only use QString, but in a case where you wanted quoting for one part of a debug line and not another, you could disable it with QDebug::noquote and then re-enable it with QDebug::quote.

Reply:
I guess disable and re-enable with quote rather than use the printable functions since that is the guidance